docs/development/workflow/development_workflow.rst

```diff
@@ -460,17 +478,49 @@ You are more likely to run into *conflicts* here--places where the changes you
 made conflict with changes that someone else made--than anywhere else. Ask for
 help if you need it.
 
-After the rebase you need to push your changes to GitHub; you will need force
-the push because `git`_ objects to re-writing the history of the repository
-after you have pushed it somewhere::
+... _howto_squash:
+
+How to squash
+=============
+
+Typically we ask to *squash*/*squish* when there was a fair amount of trial
+and error, but the final patch remains quite small, or when files were added
+and removed (especially binary files or files that should not remain in the
+repository) or if the number of commits in the history is disproportionate
+compared to the work being carried out (for example 30 commits gradually
+refining a final 10-line change).  Conceptually this is equivalent to
+exporting the final diff from a feature branch, then starting a new branch and
+only applying only that patch:
+
+The actual squashing is done on a new branch, using `git merge --squash`::
```

Comment:
I still feel that his way is weird, and more prone to mess-ups. Also what you do is basically a rebase as the new branch starts from master (btw I'm in the middle of another edit to make sure we advise people to delete their master, so better not to refer to it elsewhere in the docs, all new branches should be branched out of the remote astropy/master).

So I'm -1 on adding this section, ``git --rebase -i`` is a swiss army knife, and should be enough to show to new people here. They can still read up on other gitfu on git tutorial pages. 
However I'm asking @eteq, @astrofrog, @mhvk, @cdeil, @pllim and @Cadair about their opinion as they are the ones running into new people and teaching these steps.